src/utils/firebase.tsx

```diff
@@ -0,0 +1,54 @@
+// Import the functions you need from the SDKs you need
+import { initializeApp } from "firebase/app";
+import { getAuth, signInWithPopup, GithubAuthProvider } from "firebase/auth";
+import { firebaseConfig } from './firebaseConfig';
+
+// Initialize Firebase
+initializeApp(firebaseConfig);
+const provider = new GithubAuthProvider();
+
+const auth = getAuth();
+
+// Referenced https://firebase.google.com/docs/auth/web/github-auth
+export const signInWithGithub = async () => {
+    let token = null;
+    try {
+        const result = await signInWithPopup(auth, provider);
+        // you can access user's information with result.user
+        token = await result.user.getIdToken();
+
+        // send a request to Backend after signed up
+        fetch('http://localhost:5000/service/verify_login', {
+            headers: {
+                "Authorization": `Bearer ${token}`,
+                'Content-Type': 'application/json',
+                'Set-Cookie': 'SameSite=None; Secure;Domain=http://localhost:3000'
+            },
+            credentials: 'include'
+        })
+        .then(response => response.json())
+        .catch();
+
+    } catch (err) {
+        // Handle errors
+        console.log(err);
+        console.log("ERROR", err);
+        return { isSuccess: false};
+    }
+
+    try {
+        let res = await fetch('http://localhost:5000/service/verify_login', {
+            headers: {
+                "Authorization": `Bearer ${token}`,
+                'Content-Type': 'application/json',
+                'Set-Cookie': 'SameSite=None; Secure;Domain=http://localhost:3000'
+            },
+            credentials: 'include'
+        });
```

Comment:
Found these other areas where localhost:3000 was hardcoded and would not work in production.

Reply:
Oops, I accidentally added the same code below each other. Maybe that's the reason why you get token used too early? 

Again the localhost:3000 harcoded was left there because I was expecting to change this into environment variables in a different PR. The same logic as explained in `current_app.debug` will be applied here but in this case, it will be using procee.env.NODE_ENV passed from webpack. Let me know your thoughts about this

Reply:
Best to include this in the same PR.

Reply:
Just to let you know if we do this in this PR, I will also be updating your code in run.py line 42 as the url is also hardcoded. I still think we should create a separate PR to do this change uniformly and have a record of this in a smaller PR. If you're okay with that, then I'll just update this PR

Reply:
Just to let you know if we do this in this PR, I will also be updating your code in run.py line 42 as the url is also hardcoded. I still think we should create a separate PR to do this change uniformly and have a record of this in a smaller PR. If you're okay with that, then I'll just update this PR

Reply:
that line is a bit different (the if statement makes it not really hardcoded) but yeah we can make it a separate pr